### ISSUE TITLE
Is there a way to do a reverse slot sort in WoW?

### ISSUE THREAD

**Issue by CaspianRoach** [tullamods/Bagnon#688]
Now hear me out, what I want is not the "Reverse Bags" or "Reverse Slots" options already present in Bagnon and NOT the /run SetSortBagsRightToLeft(true) command which flips the whole thing around.

Here is what currently happens with all default settings when I click on "Clean Up" button:

![image](https://user-images.githubusercontent.com/7204157/40211480-665e3a62-5a64-11e8-9653-5d58d5fbfd3e.png)

As you can see, it sorts the items to the bottom, which is what I want, BUT it goes from the first slot of the bottom bag to the last and then goes up 1 bag. This makes for gaps and inconsisten sorting.

The thing is, if I change any settings for bag display, the new picked up items will no longer go on the top of my bags in the first slot, which is what I want.

I want the sorting and the picked up items 'meet' somewhere in the middle, so there's always some empty space between picked up items and 'sorted' items and that the sorted items have a coherent order.

The arrows show the ordering of items I want to achieve
![whatiwant](https://user-images.githubusercontent.com/7204157/40211744-cc2912bc-5a65-11e8-9988-3bddd416094d.png)

Basically, I want the sort function to do the same bag order it does now (from furthest bag to backpack), but do it in the other slot order (from last to first). Is there a way to do this with Bagnon or WoW options?

Ideally what I want is for there to exist a SetSortBagSlotsLastToFirst function or something of the sort, but it doesn't in the WoW API as far as I can see.

**Comment by CaspianRoach** [comment]
As far I can see, the extent of bag sorting code in Bagnon is calling the WoW API function SortBags() on Bagnon\external\Wildpants\components\sortButton.lua:37

and the function itself isn't documented http://wowprogramming.com/docs/api/SortBags.html

Is there a way to find out if this function accepts any arguments to tweak its logic aside from wildly putting in arguments and hoping something changes?

**Comment by TrinitysEnd** [comment]
I suggest the addon AdvancedInterfaceOptions which includes in it a reverse clean up for bags.

**Comment by CaspianRoach** [comment]
AdvancedInterfaceOptions just does /run SetSortBagsRightToLeft(true), which is not the thing I described.

**Comment by Saytix** [comment]
did you ever figure this out man? im in the same boat as you
